confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -18,133 +18,258 @@
+        """
+        Validates configuration
 
-    Arguments:
-        conf (SchemaRegistryConfig): Schema Registry config object
-    """
+        Args:
+            src_conf (dict): configuration dict
+        """
+        conf = deepcopy(src_conf)
 
-    def __init__(self, conf):
-        self._rest_client = self._RestClient(conf)
-        self.id_to_schema = {}
+        base_url = conf.get('url', '').rstrip('/')
+        if not base_url.startswith('http'):
+            raise ValueError("Invalid URL provided for Schema Registry")
 
-    def __enter__(self):
-        return self
+        ca = conf.pop('ssl.ca.location', None)
+        if ca is not None:
+            self.session.verify = ca
 
-    def __exit__(self, *args):
-        if self._rest_client is not None:
-            self._rest_client.close()
+        key = conf.pop('ssl.key.location', None)
+        cert = conf.pop('ssl.certificate.location', None)
+        if cert is not None and key is not None:
+            self.session = (key, cert)
 
-    class _RestClient(object):
-        default_headers = {'Accept': "application/vnd.schemaregistry.v1+json,"
-                                     " application/vnd.schemaregistry+json,"
-                                     " application/json"}
+        if cert is not None:
+            self.session = (key, cert)
 
-        def __init__(self, conf):
-            self.session = Session()
-            self.base_url = conf.Url.rstrip('/')
+        if key is not None and cert is None:
+            raise ValueError("ssl.key.location may not be configured without"
+                             'setting ssl.certificate.location')
 
-            if conf.Certificate is not (None, None):
-                self.session.cert = conf.Certificate
+        auth_provider = conf.pop('basic.auth.credentials.source', 'URL').upper()
+        if auth_provider not in VALID_AUTH_PROVIDERS:
+            raise ValueError("basic.auth.credentials.source must be one of"
+                             .format(VALID_AUTH_PROVIDERS))
 
-            if conf.CA is not None:
-                self.session.verify = conf.CA
+        if auth_provider == 'SASL_INHERIT':
+            if conf.pop('sasl.mechanism', '').upper() == 'GSSAPI':
+                raise ValueError("SASL_INHERIT does not support SASL mechanism GSSAPI")
+            self.session.auth = (conf.pop('sasl.username', ''),
+                                 conf.pop('sasl.password', ''))
 
-            if conf.Credentials:
-                self.session.auth = conf.Credentials
+        if auth_provider == 'USER_INFO':
+            self.session.auth = tuple(conf.pop('basic.auth.user.info', '').split(':'))
+        else:
+            self.session.auth = utils.get_auth_from_url(base_url)
 
-        def __del__(self):
-            self.session.close()
+        return conf
 
-        def GET(self, url, decoder=None):
-            return self.send_request(url, method='GET', decoder=decoder)
+    def close(self):
```

Comment:
since session cares enough to have a `close` method i think we should definitely care enough to transfer this through to the SR client. also, I think if a class has a close, it's sloppy programming not to call it.